Makefile

```diff
@@ -46,12 +47,14 @@ endef
 ifeq ($(HELP),y)
 all: clean
 	@echo "$$ALL_HELP_INFO"
+else ifeq ($(KUBEEDGE_COMPILE_WITH_IMAGE),true)
+all:
+	hack/make-rules/build.sh $(WHAT)
```

Comment:
can also delete this block directly, or emphasize the env `KUBEEDGE_COMPILE_WITH_IMAGE` to make it separated from using local go. ?

Reply:
`BUILD_WITH_CONTAINER` will be a better name, in container, not image.

Reply:
We can set `BUILD_WITH_CONTAINER` as a env value like istio? so no need to add it to cmd everytime.

Reply:
Looks like adding another script for container is fine.

Reply:
```
root@ecs-8f95:~/go/src/github.com/kubeedge/kubeedge# export BUILD_WITH_CONTAINER=true
root@ecs-8f95:~/go/src/github.com/kubeedge/kubeedge# make all WHAT=keadm
hack/make-rules/build_with_container.sh keadm
building github.com/kubeedge/kubeedge/keadm/cmd/keadm
+ docker run --rm -v /root/go/src/github.com/kubeedge/kubeedge:/kubeedge -w /kubeedge kubeedge/build-tools /usr/local/go/bin/go build -o /kubeedge/_output/local/bin/keadm -gcflags= -ldflags '-s -w -buildid= -X github.com/kubeedge/kubeedge/pkg/version.buildDate=2022-04-02T00:58:49Z -X github.com/kubeedge/kubeedge/pkg/version.gitCommit=e0d10fd6800e0fb02a9c307d98c23eac3d589121 -X github.com/kubeedge/kubeedge/pkg/version.gitTreeState=dirty -X github.com/kubeedge/kubeedge/pkg/version.gitVersion=v1.5.0-beta.0.1359+e0d10fd6800e0f-dirty -X github.com/kubeedge/kubeedge/pkg/version.gitMajor=1 -X github.com/kubeedge/kubeedge/pkg/version.gitMinor=5+' github.com/kubeedge/kubeedge/keadm/cmd/keadm
+ set +x
root@ecs-8f95:~/go/src/github.com/kubeedge/kubeedge# unset BUILD_WITH_CONTAINER
root@ecs-8f95:~/go/src/github.com/kubeedge/kubeedge# make all WHAT=keadm
KUBEEDGE_OUTPUT_SUBPATH=_output/local hack/make-rules/build.sh keadm
go detail version: go version go1.16.15 linux/amd64
go version: 1.16.15
building github.com/kubeedge/kubeedge/keadm/cmd/keadm
+ go build -o /root/go/src/github.com/kubeedge/kubeedge/_output/local/bin/keadm -gcflags= -ldflags '-s -w -buildid= -X github.com/kubeedge/kubeedge/pkg/version.buildDate=2022-04-02T01:00:08Z -X github.com/kubeedge/kubeedge/pkg/version.gitCommit=e0d10fd6800e0fb02a9c307d98c23eac3d589121 -X github.com/kubeedge/kubeedge/pkg/version.gitTreeState=dirty -X github.com/kubeedge/kubeedge/pkg/version.gitVersion=v1.5.0-beta.0.1359+e0d10fd6800e0f-dirty -X github.com/kubeedge/kubeedge/pkg/version.gitMajor=1 -X github.com/kubeedge/kubeedge/pkg/version.gitMinor=5+' github.com/kubeedge/kubeedge/keadm/cmd/keadm
+ set +x
```

Reply:
And can also specify the BUILD_WITH_CONTAINER env in make command, just like this:
```
root@ecs-8f95:~/go/src/github.com/kubeedge/kubeedge# make all WHAT=keadm BUILD_WITH_CONTAINER=true
hack/make-rules/build_with_container.sh keadm
building github.com/kubeedge/kubeedge/keadm/cmd/keadm
+ docker run --rm -v /root/go/src/github.com/kubeedge/kubeedge:/kubeedge -w /kubeedge kubeedge/build-tools /usr/local/go/bin/go build -o /kubeedge/_output/local/bin/keadm -gcflags= -ldflags '-s -w -buildid= -X github.com/kubeedge/kubeedge/pkg/version.buildDate=2022-04-02T01:22:49Z -X github.com/kubeedge/kubeedge/pkg/version.gitCommit=b216deb58ac6de7f1e5bcab9c3ec136286ae3647 -X github.com/kubeedge/kubeedge/pkg/version.gitTreeState=clean -X github.com/kubeedge/kubeedge/pkg/version.gitVersion=v1.5.0-beta.0.1357+b216deb58ac6de -X github.com/kubeedge/kubeedge/pkg/version.gitMajor=1 -X github.com/kubeedge/kubeedge/pkg/version.gitMinor=5+' github.com/kubeedge/kubeedge/keadm/cmd/keadm
+ set +x



root@ecs-8f95:~/go/src/github.com/kubeedge/kubeedge# make all WHAT=keadm 
KUBEEDGE_OUTPUT_SUBPATH=_output/local hack/make-rules/build.sh keadm
go detail version: go version go1.16.15 linux/amd64
go version: 1.16.15
building github.com/kubeedge/kubeedge/keadm/cmd/keadm
+ go build -o /root/go/src/github.com/kubeedge/kubeedge/_output/local/bin/keadm -gcflags= -ldflags '-s -w -buildid= -X github.com/kubeedge/kubeedge/pkg/version.buildDate=2022-04-02T01:23:51Z -X github.com/kubeedge/kubeedge/pkg/version.gitCommit=b216deb58ac6de7f1e5bcab9c3ec136286ae3647 -X github.com/kubeedge/kubeedge/pkg/version.gitTreeState=clean -X github.com/kubeedge/kubeedge/pkg/version.gitVersion=v1.5.0-beta.0.1357+b216deb58ac6de -X github.com/kubeedge/kubeedge/pkg/version.gitMajor=1 -X github.com/kubeedge/kubeedge/pkg/version.gitMinor=5+' github.com/kubeedge/kubeedge/keadm/cmd/keadm
+ set +x
```